packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -38,30 +40,103 @@ class _TextSelectionToolbar extends StatelessWidget {
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  @override
+  __TextSelectionToolbarState createState() => __TextSelectionToolbarState();
+}
+
+class __TextSelectionToolbarState extends State<_TextSelectionToolbar> {
+  // effective when items > 3
+  bool flipToMenu = false;
```

Comment:
This should be private if it's not used externally.

Maybe something like this:

```dart
// Any menu items that overflow the screen are placed into a
// submenu. If this happens, this bool indicates whether or not
// the menu is open.
bool _submenuIsOpen = false;
```